examples/wasm/methods/guest/src/main.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_main]
+#![allow(unused_imports)]
+
+risc0_zkvm::guest::entry!(main);
+
+use externc_libm::math::fmod;
+use risc0_zkvm::guest::env;
+use wasmi::{Caller, Engine, Func, Linker, Module, Store};
+
+pub fn main() {
+    let engine = Engine::default();
+
+    let wasm: Vec<u8> = env::read();
+    let iters: i32 = env::read();
+
+    // Derived from the wasmi example: https://docs.rs/wasmi/0.29.0/wasmi/#example
+    let module = Module::new(&engine, &mut &wasm[..]).expect("Failed to create module");
+    type HostState = u32;
+
+    let linker = <Linker<HostState>>::new(&engine);
+    let mut store = Store::new(&engine, 42);
```

Comment:
Why does this store `42`? This feels like it may be an accidental relic of the source example...

Reply:
Yeah I think its an artifact of https://docs.rs/wasmi/latest/wasmi/#example and it requires some data to be passed with the correct type: https://docs.rs/wasmi/latest/wasmi/struct.Store.html#method.new

Reply:
:+1: gotcha, so the value is arbitrary and unused, but having _something_ there is required.